nevergrad/optimization/optimizerlib.py

```diff
@@ -247,6 +255,12 @@ def _internal_ask_candidate(self) -> p.Parameter:
                     "portfolio": mutator.portfolio_discrete_mutation,
                 }[mutation]
                 data = func(pessimistic_data, arity=self.arity_for_discrete_mutation)
+            if self.sparse:
+                data = np.asarray(data)
+                zeroing = (
+                    np.asarray(np.random.randint(data.size + 1, size=data.size).reshape(data.shape)) == 0
+                )
```

Comment:
Why use asarray? It should already be one
```suggestion
                zeroing = np.random.randint(data.size + 1, size=data.size).reshape(data.shape)) == 0
```
Also, is it intentional that in some cases nothing will change?